### PR TITLE
new: `get/put /_admin/cluster/maintenance/`

### PR DESCRIPTION
**Pull request by aMahanna**

### CODE REVIEW

File: tests/test_cluster.py [hunk 1]
```diff
@@ -112,6 +113,28 @@ def test_cluster_server_statistics(sys_db, bad_db, cluster):
...
+
+    server_id = sys_db.cluster.server_id()
+    result = sys_db.cluster.server_maintenance_mode(server_id)
+    assert result == {}
+
+    with assert_raises(ClusterMaintenanceModeError) as err:
+        bad_db.cluster.server_maintenance_mode(server_id)
+    assert err.value.error_code in {FORBIDDEN, DATABASE_NOT_FOUND}
+
+    sys_db.cluster.toggle_server_maintenance_mode(server_id, "maintenance", timeout=1)
```
Comment (aMahanna) [inline tests/test_cluster.py:128]: causing read timeout issues on _some_ of the cluster instances

Comment (apetenchea) [reply tests/test_cluster.py:137]: Tried to use a dbserver instead, but that didn't fix the issue.

Comment (aMahanna) [reply tests/test_cluster.py:137]: yeah really strange timeout, I don't have anything in mind yet

Comment (aMahanna) [reply tests/test_cluster.py]: thanks for the fix @apetenchea

**Pull request merged**